assets/js/state/user.js

```diff
@@ -0,0 +1,36 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+// Fields set to undefined for the sake of documenting the state shape
+export const initialState = {
+  loggedIn: false,
+  username: undefined,
+  authError: null,
+  authInProgress: false,
+};
+
+export const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    setAuthInProgress(state, _payload) {
+      state.authInProgress = true;
+    },
+    setAuthError(state, { payload: { error } }) {
+      state.authInProgress = false;
+      state.authError = error;
+    },
+    setUserAsLogged(state, _payload) {
+      state.loggedIn = true;
+      state.authInProgress = false;
+      state.authError = null;
+    },
+    setUser(state, { payload: { username } }) {
```

Comment:
Why do we need an independent `setUser` function? Couldn't we put it in `setUserAsLogged` and pass the username there?

Reply:
Yes, we can but I prefer to distinguish the two things and have two separate action, in order to attach sagas to both of the authentication actions if needed, but we can discuss about that